DataManagementSystem/Client/FailoverTransfer.py

```diff
@@ -232,32 +248,32 @@ def _setRegistrationRequest( self, lfn, targetSE, fileDict, catalog ):
 
       regFile = File()
       regFile.LFN = lfn
-      regFile.Checksum = fileDict.get( "Checksum", "" )
-      regFile.ChecksumType = fileDict.get( "ChecksumType", self.defaultChecksumType )
-      regFile.Size = fileDict.get( "Size", 0 )
-      regFile.GUID = fileDict.get( "GUID", "" )
+      regFile.Checksum = fileDict.get("Checksum", "")
+      regFile.ChecksumType = fileDict.get("ChecksumType", self.defaultChecksumType)
+      regFile.Size = fileDict.get("Size", 0)
+      regFile.GUID = fileDict.get("GUID", "")
 
-      se = StorageElement( targetSE )
-      pfn = se.getURL( lfn, self.registrationProtocols )
-      if not pfn["OK"] or lfn not in pfn["Value"]['Successful']:
-        self.log.error( "Unable to get PFN for LFN", "%s" % pfn.get( 'Message', pfn.get( 'Value', {} ).get( 'Failed', {} ).get( lfn ) ) )
-        return pfn
-      regFile.PFN = pfn["Value"]['Successful'][lfn]
+      se = StorageElement(targetSE)
+      res = returnSingleResult(se.getURL(lfn, self.registrationProtocols))
+      if not res["OK"]:
+        self.log.error("Unable to get PFN for LFN", res['Message'])
+        return res
+      regFile.PFN = res["Value"][lfn]
```

Comment:
I think this has to be
```suggestion
      regFile.PFN = res["Value"]
```

Reply:
Right, corrected.